### DISCUSSION TITLE
Do syntax highlighting on large files

### DISCUSSION THREAD

**Discussion by oarcelus** [nvim-treesitter/nvim-treesitter#6227] [General]
TreeSitter offers a way of disabling syntax highlighting based on file size. This will make the editing faster but it may ruin the experience for some users. Would it be possible to apply the syntax highlighting only for a particular 'band' of lines centered around the current cursor position. Some sort of padding around the cursor positions in which the syntax highlighting is applied is what I am talking about. Would this make it faster?

**Comment by clason** [comment]
No. The parsing needs to be global, since the tree structure can depend on anything anywhere. And the highlights are already applied for the visible window only. (Injections you only need to run for the viewport, too, but that is also already done.)
